### PR TITLE
bpo-35640: Allow passing PathLike arguments to SimpleHTTPRequestHandler

### PR DESCRIPTION
**Pull request by geryogam**
https://bugs.python.org/issue35640

### CODE REVIEW

**Comment by tirkarthi** [comment]
Please raise an issue in https://bugs.python.org for this PR to discuss this since it's a non-trivial change and would require tests.

**Comment by eamanu** [comment]
This seems a interest. I open a issue https://bugs.python.org/issue35640 to discuss this @tirkarthi  @maggyero

**Review by eamanu — APPROVED** [review]
LGTM. @maggyero Maybe you will need some Exception control.

**Comment by JulienPalard** [comment]
Hi @maggyero thanks for the PR!

Could you also provide a `.. versionchanged::` in the documentation to tell about it, and a test?

**Comment by maggyero** [comment]
> Could you also provide a `.. versionchanged::` in the documentation to tell about it, and a test?

@JulienPalard Yes, done.

**Comment by maggyero** [comment]
> Could you also provide a `.. versionchanged::` in the documentation to tell about it, and a test?

@JulienPalard Yes, done.

**Comment by maggyero** [comment]
@JulienPalard Up.

**Comment by maggyero** [comment]
@JulienPalard I have made the requested changes. Please review again.

**Comment by maggyero** [comment]
@JulienPalard

**Comment by maggyero** [comment]
@JulienPalard I have made the requested changes. Please review again.

**Comment by zware** [comment]
@maggyero Could you add a NEWS entry for this change?

**Comment by maggyero** [comment]
@zware Yes, done!

File: Doc/library/http.server.rst [hunk 1]
```diff
@@ -344,6 +344,9 @@ provides three different variants:
 
       If not specified, the directory to serve is the current working directory.
 
+      .. versionchanged:: 3.7
```
Comment (zware) [inline Doc/library/http.server.rst]:
```suggestion
      .. versionchanged:: 3.9
```

**Review by zware — CHANGES_REQUESTED** [review]
One last thing that I'll commit myself if GitHub lets me, otherwise LGTM.

File: Doc/library/http.server.rst (hunk 1 above)
Comment (zware) [reply Doc/library/http.server.rst]: It looks like GitHub won't let me commit this myself :)

Comment (geryogam) [reply Doc/library/http.server.rst]: Done, thanks!

**Comment by maggyero** [comment]
I have made the requested changes; please review again.

**Review by zware — APPROVED** [review]
LGTM!

**Comment by zware** [comment]
Thanks for the patch, @maggyero!

**Comment by maggyero** [comment]
Thanks for the review @zware!

**Pull request merged**